hi-ml-azure/testazure/testazure/test_azure_util.py

```diff
@@ -811,35 +812,123 @@ def test_set_environment_variables_for_multi_node(
         caplog: LogCaptureFixture,
         capsys: CaptureFixture,
 ) -> None:
+    # If none of AZ_BATCHAI_MPI_MASTER_NODE, AZ_BATCH_MASTER_NODE or ENV_MASTER_IP are set, should assume
+    # single node training job
     with caplog.at_level(logging.INFO):  # type: ignore
         util.set_environment_variables_for_multi_node()
-        assert "No settings for the MPI central node found" in caplog.text  # type: ignore
+        assert "No settings for the MPI central node found" in caplog.messages[-1]   # type: ignore
         assert "Assuming that this is a single node training job" in caplog.text  # type: ignore
 
-    with mock.patch.dict(
-            os.environ,
-            {
-                util.ENV_AZ_BATCHAI_MPI_MASTER_NODE: "here",
-                util.ENV_MASTER_PORT: "there",
-                util.ENV_OMPI_COMM_WORLD_RANK: "everywhere",
-                util.ENV_MASTER_ADDR: "else",
-            },
-            clear=True):
-        util.set_environment_variables_for_multi_node()
-    out = capsys.readouterr().out
-    assert "Distributed training: MASTER_ADDR = here, MASTER_PORT = there, NODE_RANK = everywhere" in out
+    # If all of ENV_MASTER_IP, AZ_BATCHAI_MPI_MASTER_NODE and AZ_BATCH_MASTER_NODE are set, the latter should
+    # take precedence. NODE_RANK should get updated to the value of ENV_OMPI_COMM_WORLD_RANK
+    mock_az_batch_master_port = str(randint(1000, MASTER_PORT_DEFAULT - 1))  # Avoid setting to the default value
+    mock_node_rank = str(randint(0, 10))
+
+    mock_az_batch_master_addr = "1234.0.0.0"
+    mock_az_batch_master_node = f"{mock_az_batch_master_addr}:{mock_az_batch_master_port}"
+
+    mock_az_batch_mpi_master_addr = "5678.9.9.9"
+    mock_az_batch_mpi_master_node = f"{mock_az_batch_mpi_master_addr}"
```

Comment:
Nitpick: I'm having a hard time reading the tests because many variables have the same prefix - causing the same confusion that we had yesterday when analyzing the problem. How about simplifying the names to be something like "master", "master_and_port", "port"?